python/rikai/io.py

```diff
@@ -76,17 +109,19 @@ def copy(source: str, dest: str) -> str:
 
     if parsed_dest.scheme == parsed_source.scheme:
         # Direct copy with the same file system
-        if parsed_dest.scheme == "s3":
+        scheme = parsed_dest.scheme
+        if scheme == "s3":
             s3fs, source_path = fs.FileSystem.from_uri(source)
             _, dest_path = fs.FileSystem.from_uri(dest)
             s3fs.copy(source_path, dest_path)
             return dest
+        elif scheme == "gs":
+            _gcsfs().copy(source, dest)
+            return dest
 
     # TODO: find better i/o utilis to copy between filesystems
-    filesystem, dest_path = fs.FileSystem.from_uri(dest)
-    with filesystem.open_output_stream(dest_path) as out_stream:
-        src_fs, src_path = fs.FileSystem.from_uri(source)
-        with src_fs.open_input_stream(src_path) as in_stream:
+    with _open_output_stream(dest) as out_stream:
+        with _open_input_stream(source) as in_stream:
             while True:
                 buf = in_stream.read(_BUFSIZE)
```

Comment:
Can probably use `shutil` here instead of implementing it ourselves.

Reply:
Can shutil work with cloud storage?

Reply:
I think [`shutil.copyfileobj`](https://docs.python.org/3/library/shutil.html#shutil.copyfileobj) can work with any file-like object.

Reply:
`shutil.copyfileobj` ?